Gems/ROS2/Code/Source/Camera/CameraPublishers.cpp

```diff
@@ -0,0 +1,122 @@
+
+        template<>
+        TopicConfigurations GetCameraInfoTopicConfiguration<CameraDepthSensor>(const SensorConfiguration& sensorConfiguration)
+        {
+            return { { CameraPublishers::DEPTH, sensorConfiguration.m_publishersConfigurations[CameraConstants::DepthInfoConfig] } };
+        }
+
+        //! Helper that adds publishers based on predefined configuration.
+        template<typename PublishedData>
+        void AddPublishersFromConfiguration(
+            const TopicConfigurations configurations,
+            AZStd::unordered_map<CameraSensorDescription::CameraChannelType, std::shared_ptr<rclcpp::Publisher<PublishedData>>>& publishers)
+        {
+            for (const auto& [channel, configuration] : configurations)
+            {
+                AZStd::string fullTopic = ROS2Names::GetNamespacedName(GetNamespace(), configuration.m_topic);
+                auto ros2Node = ROS2Interface::Get()->GetNode();
+                auto publisher = ros2Node->create_publisher<PublishedData>(fullTopic.data(), configuration.GetQoS());
+                publishers[channel] = publisher;
+            }
+        }
+
+        //! Helper that adds publishers for a camera type.
+        //! @tparam CameraType type of camera sensor (eg 'CameraColorSensor')
+        template<typename CameraType>
+        void AddCameraPublishers(
+            const SensorConfiguration& sensorConfiguration,
+            AZStd::unordered_map<CameraSensorDescription::CameraChannelType, ImagePublisherPtrType>& imagePublishers,
+            AZStd::unordered_map<CameraSensorDescription::CameraChannelType, InfoPublisherPtrType>& infoPublishers)
+        {
+            const auto cameraImagePublisherConfigs = GetCameraTopicConfiguration<CameraType>(sensorConfiguration);
+            AddPublishersFromConfiguration(cameraImagePublisherConfigs, imagePublishers);
+            const auto cameraInfoPublisherConfigs = GetCameraInfoTopicConfiguration<CameraType>(sensorConfiguration);
+            AddPublishersFromConfiguration(cameraInfoPublisherConfigs, cameraInfoPublishers);
+        }
+    } // namespace Internal
+
+    CameraPublishers::CameraPublishers(const CameraSensorDescription& cameraDescription)
+    {
+        if (cameraDescription.m_cameraConfiguration.m_colorCamera)
+        {
+            Internal::AddCameraPublishers<CameraColorSensor>(cameraDescription.m_sensorConfiguration, m_imagePublishers, m_infoPublishers);
+        }
+
+        if (cameraDescription.m_cameraConfiguration.m_depthCamera)
+        {
+            Internal::AddCameraPublishers<CameraDepthSensor>(cameraDescription.m_sensorConfiguration, m_imagePublishers, m_infoPublishers);
+        }
+    }
+
+    ImagePublisherPtrType CameraPublishers::GetImagePublisher(ChannelType type)
+    {
+        if (m_imagePublishers.count(type) != 1)
+        {
+            AZ_Error("GetImagePublisher", false, "No publisher of this type, logic error!");
+        }
+        return m_imagePublishers.at(type);
```

Comment:
You can also make use of the 2nd parameter for `AZ_Error`
```suggestion
        // There should be exactly one image publisher of type `type`
        AZ_Error("GetImagePublisher", (m_imagePublishers.count(type) == 1), "No publisher of this type, logic error!");

        return m_imagePublishers.at(type);
```